app/code/Magento/ConfigurableProductGraphQl/Model/Options/Collection.php

```diff
@@ -87,6 +122,22 @@ public function getAttributesByProductId(int $productId) : array
             return [];
         }
 
+        $product = $this->productRepository->getById($productId);
```

Comment:
NoSuchEntityException is not handled, getting internal server error when querying variants and configurable_options. Might be breaking current tests.